File: backend/internal/model/recognition.go
package model

// RecognitionApplication 家庭经济困难学生认定申请表
type RecognitionApplication struct {
	BaseModel
	StudentID uint `gorm:"index;not null" json:"student_id"`
	Year      int  `gorm:"index;not null" json:"year"` // 认定学年

	// 基本情况
	Nation           string `gorm:"size:32" json:"nation"`
	NativePlace      string `gorm:"size:128" json:"native_place"` // 籍贯（到市/县）
	IDCard           string `gorm:"size:18" json:"id_card"`
	FamilyPopulation int    `json:"family_population"` // 家庭人口
	Phone            string `gorm:"size:20" json:"phone"`
	Address          string `gorm:"size:255" json:"address"` // 详细通讯地址
	PostalCode       string `gorm:"size:16" json:"postal_code"`
	GuardianPhone    string `gorm:"size:20" json:"guardian_phone"`

	// 家庭经济情况
	HouseholdType         HouseholdType `gorm:"size:16" json:"household_type"` // 城镇/农村
	PerCapitaAnnualIncome float64       `json:"per_capita_annual_income"`      // 家庭人均年收入
	IncomeSource          string        `gorm:"size:128" json:"income_source"` // 收入来源

	// 特殊群体勾选（逗号分隔的 SpecialGroupType 集合）
	SpecialTypes string `gorm:"size:255" json:"special_types"`

	// 影响家庭经济状况有关信息（无则填"无"）
	NaturalDisaster string `gorm:"size:512" json:"natural_disaster"` // 自然灾害
	SuddenAccident  string `gorm:"size:512" json:"sudden_accident"`  // 突发意外事件
	WeakLabor       string `gorm:"size:512" json:"weak_labor"`       // 残疾/年迈劳动力弱
	Unemployment    string `gorm:"size:512" json:"unemployment"`     // 失业情况
	Debt            string `gorm:"size:512" json:"debt"`             // 欠债情况
	OtherInfo       string `gorm:"size:1024" json:"other_info"`      // 其他情况

	// 个人承诺（线上勾选记录，签字线下手写）
	CommitmentAgreed bool `gorm:"default:false" json:"commitment_agreed"`

	// 流程与结果
	Status          ApplicationStatus `gorm:"size:32;index;default:'draft'" json:"status"`
	CurrentLevel    ReviewLevel       `gorm:"default:0" json:"current_level"`  // 当前所处评审级别
	DifficultyLevel DifficultyLevel   `gorm:"size:16" json:"difficulty_level"` // 最终困难等级
	RejectReason    string            `gorm:"size:512" json:"reject_reason"`

	FamilyMembers []FamilyMember `gorm:"foreignKey:ApplicationID" json:"family_members,omitempty"`
	Reviews       []ReviewRecord `gorm:"foreignKey:ApplicationID" json:"reviews,omitempty"`
}

// FamilyMember 家庭成员
type FamilyMember struct {
	BaseModel
	ApplicationID uint    `gorm:"index;not null" json:"application_id"`
	Name          string  `gorm:"size:64" json:"name"`
	Age           int     `json:"age"`
	Relation      string  `gorm:"size:32" json:"relation"`     // 与学生关系（关系，非称呼）
	WorkUnit      string  `gorm:"size:255" json:"work_unit"`   // 工作/学习单位
	Occupation    string  `gorm:"size:16" json:"occupation"`   // 务工/务农/无/读书
	AnnualIncome  float64 `json:"annual_income"`               // 年收入
	Health        string  `gorm:"size:16" json:"health"`       // 良好/较差/残疾
	SpecialType   string  `gorm:"size:32" json:"special_type"` // 特殊群体类型
}

// ReviewRecord 评审记录（流转日志/审计）
type ReviewRecord struct {
	BaseModel
	ApplicationID   uint            `gorm:"index;not null" json:"application_id"`
	Level           ReviewLevel     `gorm:"index" json:"level"`
	ReviewerID      uint            `gorm:"index" json:"reviewer_id"`
	Action          ReviewAction    `gorm:"size:16" json:"action"`
	Opinion         string          `gorm:"size:512" json:"opinion"`
	DifficultyLevel DifficultyLevel `gorm:"size:16" json:"difficulty_level"`  // 该级定/调整的困难等级
	RejectToLevel   ReviewLevel     `gorm:"default:0" json:"reject_to_level"` // 退回到哪一级（0=退回学生）
}

File: frontend/src/types/recognition.ts
// 困难认定申请类型（与 backend/internal/dto/recognition.go 对齐）

import type { PageResult } from "@/types/student";

export type { PageResult };

// 申请状态（与 backend/internal/model/enums.go ApplicationStatus 对齐）
export type ApplicationStatus =
  | "draft"
  | "pending_class"
  | "pending_dept"
  | "pending_college"
  | "pending_final"
  | "approved"
  | "rejected";

// 困难等级（与 difficulty_level 字典对齐，未评定为空串）
export type DifficultyLevel = "" | "special" | "hard" | "general";

// 户口类型
export type HouseholdType = "" | "urban" | "rural";

// 家庭成员录入项
export interface FamilyMemberInput {
  name: string;
  age: number;
  relation: string; // 字典 relation
  work_unit: string;
  occupation: string; // 字典 occupation
  annual_income: number;
  health: string; // 字典 health_status
  special_type: string; // 特殊群体类型（可空）
}

// 家庭成员响应
export interface FamilyMember extends FamilyMemberInput {
  id: number;
}

// 认定申请创建/修改请求（草稿与提交共用）
export interface RecognitionInput {
  year: number;

  nation: string;
  native_place: string;
  id_card: string;
  family_population: number;
  phone: string;
  address: string;
  postal_code: string;
  guardian_phone: string;

  household_type: string;
  per_capita_annual_income: number;
  income_source: string;

  special_types: string[];

  natural_disaster: string;
  sudden_accident: string;
  weak_labor: string;
  unemployment: string;
  debt: string;
  other_info: string;

  commitment_agreed: boolean;

  family_members: FamilyMemberInput[];
}

// 评审动作
export type ReviewActionType = "pass" | "reject";

// 评审流转记录（与 backend/internal/dto/review.go ReviewRecordResponse 对齐）
export interface ReviewRecord {
  id: number;
  level: number;
  reviewer_id: number;
  reviewer_name: string;
  action: ReviewActionType | string;
  opinion: string;
  difficulty_level: DifficultyLevel | string;
  reject_to_level: number;
  created_at: string;
}

// 认定申请详情响应
export interface Recognition extends RecognitionInput {
  id: number;
  student_id: number;
  student_no: string;
  student_name: string;
  dept_name: string;
  class_name: string;
  status: ApplicationStatus;
  current_level: number;
  difficulty_level: DifficultyLevel;
  reject_reason: string;
  family_members: FamilyMember[];
  reviews: ReviewRecord[];
}

// 单条评审动作请求
export interface ReviewActionInput {
  difficulty_level?: string;
  opinion?: string;
  reject_to_level?: number;
}

// 批量评审请求
export interface BatchReviewInput {
  ids: number[];
  action: ReviewActionType;
  difficulty_level?: string;
  opinion?: string;
  reject_to_level?: number;
}

// 批量评审单条结果
export interface BatchReviewItemResult {
  id: number;
  ok: boolean;
  message?: string;
}

// 批量评审汇总结果
export interface BatchReviewResult {
  total: number;
  success: number;
  failed: number;
  items: BatchReviewItemResult[];
}

// 认定申请列表项
export interface RecognitionListItem {
  id: number;
  student_id: number;
  student_no: string;
  student_name: string;
  dept_name: string;
  major_name: string;
  class_name: string;
  year: number;
  status: ApplicationStatus;
  current_level: number;
  difficulty_level: DifficultyLevel;
  per_capita_annual_income: number;
  special_types: string[];
  /** 低收入证明材料份数（不含签字图）。 */
  proof_count: number;
}

// 提交结果：申请快照 + 非阻断性提示
export interface SubmitResult {
  application: Recognition;
  warnings: string[];
}

// 附件响应
export interface Attachment {
  id: number;
  owner_type: string;
  owner_id: number;
  file_name: string;
  size: number;
  mime: string;
  uploader_id: number;
}

// 列表筛选
export interface RecognitionFilter {
  page?: number;
  page_size?: number;
  year?: number;
  status?: string;
  keyword?: string;
  special_type?: string;
  dept_id?: number;
  class_id?: number;
  tab?: "todo" | "done" | "all";
  ids?: number[];
  scope?: "todo" | "approved";
}
